src/daemon/plugin.c

```diff
@@ -2801,8 +2832,22 @@ int plugin_thread_create (pthread_t *thread, const pthread_attr_t *attr,
 	plugin_thread->start_routine = start_routine;
 	plugin_thread->arg           = arg;
 
-	return pthread_create (thread, attr,
+	ret = pthread_create (thread, attr,
 			plugin_thread_start, plugin_thread);
+
+	if (ret == 0 && name != NULL) {
+#if defined(HAVE_PTHREAD_SETNAME_NP) || defined(HAVE_PTHREAD_SET_NAME_NP)
+		char thread_name[16];
+		sstrncpy (thread_name, name, sizeof(thread_name));
```

Comment:
Indeed, Linux / glibc even returns `ERANGE` if the name it too long. This should be commented in the code.